host.c

```diff
@@ -248,8 +252,26 @@ void SYSINFO_Init(void)
 #elif defined(__APPLE__)
 void SYSINFO_Init(void)
 {
-	// TODO: disconnect --> f_system for MacOSX (man sysctl)
-	// VVD: Look at code for FreeBSD: 30 lines down. :-)
+    int mib[2];
+    mib[0] = CTL_HW;
+    mib[1] = HW_MEMSIZE;
+    int64_t memsize_value;
+    size_t length = sizeof(memsize_value);
+    if (sysctl(mib, 2, &memsize_value, &length, NULL, 0) != -1)
+    	SYSINFO_memory = memsize_value;
+
+    char cpu_brand_string[100];
+    size_t cpu_brand_string_len = 100;
+    if (sysctlbyname("machdep.cpu.brand_string", &cpu_brand_string, &cpu_brand_string_len, NULL, 0) != -1)
+        SYSINFO_processor_description = cpu_brand_string;
+
+    mib[0] = CTL_HW;
+    mib[1] = HW_CPU_FREQ;
+    int cpu_frequency_value;
```

Comment:
Haven't touched the ezquake code in a long while, but as I remember, the code was written in C89 compatible style, so that variables are declared at the beginning of the scope always. Maybe I remember wrong, I'm just used to writing all my code like that for max portability.